File: 02 medium/array/56_merge_intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b) => a[0] - b[0])
    let result = [intervals[0]]
    for (var i = 0; i < intervals.length; i++) {
        let res = result[result.length - 1] 
        if (intervals[i][0] <= res[1]) result[result.length - 1][1] = Math.max(intervals[i][1], res[1])
        else result.push(intervals[i])
    }
    return result;
};

File: array/56_merge_intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        
        intervals = sorted(intervals)
        result = []

        curr_max = intervals[0][1]
        curr_min = intervals[0][0]
        
        for interval in intervals:
            if interval[0] <= curr_max:
                curr_max = max(curr_max, interval[1])
            else:
                result.append([curr_min, curr_max])
                curr_min = interval[0]
                curr_max = interval[1]

        result.append([curr_min, curr_max])

        print(result)
        return result
